thingif/src/test/java/com/kii/thingif/thingifapi/BuilderTest.java

```diff
@@ -0,0 +1,149 @@
+        Assert.assertEquals(api.getStateTypes(), getDefaultStateTypes());
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullContextTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                null, 
+                new KiiApp("appid", "appkey", Site.JP), 
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullAppIDTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp(null, "appkey", Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithEmptyAppIDTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("", "appkey", Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullAppKeyTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", null, Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithEmptyAppKeyTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", "", Site.JP),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullSiteTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", "appkey", (Site) null),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullOwnerTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", "appkey", Site.JP), null);
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithNullBaseUrlTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", "appkey", (String)null),
+                new Owner(new TypedID(TypedID.Types.USER, "user1234"), "token"));
+    }
+    @Test(expected = IllegalArgumentException.class)
+    public void newBuilderWithEmptyBaseUrlTest() throws Exception {
+        ThingIFAPI.Builder.newBuilder(
+                this.context,
+                new KiiApp("appid", "appkey", ""),
```

Comment:
This is Test for `KiiApp` constructor. `ThingIFAPI.Builder` is needless.

What do you think?